lastseen.py

```diff
@@ -0,0 +1,87 @@
+from itertools import chain
+from datetime import datetime
+from errbot import BotPlugin, botcmd
+
+USAGE_STR = 'Nick not specified. (usage: !last_seen <nick>)'
+CONFIG_TEMPLATE = {
+    # locale-preffered format is default
+    'TIME_FORMAT': '%c',
+    'MAX_CHARS': 40
+}
+
+
+class LastSeen(BotPlugin):
+    def configure(self, configuration):
+        if configuration is not None and configuration != {}:
+            config = dict(chain(CONFIG_TEMPLATE.items(),
+                                configuration.items()))
+        else:
+            config = CONFIG_TEMPLATE
+        super(LastSeen, self).configure(config)
+
+    def get_configuration_template(self):
+        return CONFIG_TEMPLATE
+
+    def activate(self):
+        super(LastSeen, self).activate()
+        try:
+            if type(self['last_seen']) is not dict:
+                self['last_seen'] = {}
+        except KeyError:
+            self['last_seen'] = {}
+
+    def callback_message(self, msg):
+        user = msg.nick
+        last_seens = self['last_seen']
+
+        last_seens[user] = {}
+        last_seens[user]['time'] = datetime.now()
+        last_seens[user]['msg'] = msg.body
+
+        self['last_seen'] = last_seens
+
+    def get_msg(self, timedelta, user, text):
```

Comment:
@JakubNvk Could `get_msg` be replaced with something more descriptive? Currently its name does not say much about what it does. Maybe something like `generate_seen_message(timedelta, user, text)`?